### PR TITLE
-Wconversion fixes

### PR DESCRIPTION
**Pull request by sambrg**

### CODE REVIEW

**Comment by gracicot** [comment]
Would there be a way to do that without macros? Or at least protect the macros sufficiently with a prefix?

**Comment by sambrg** [comment]
Apart from expanding the macro not that I know. Better named macro should be better, perhaps BC_SET_ARGS, is specific enough?

File: include/inja/parser.hpp [hunk 1]
```diff
@@ -300,11 +300,11 @@ class Parser {
 
       // end previous block with unconditional jump to endif; destination will be
       // filled in by endif
-      if_data.uncond_jumps.push_back(tmpl.bytecodes.size());
+      if_data.uncond_jumps.push_back((uint32_t)tmpl.bytecodes.size());
```
Comment (gracicot) [inline include/inja/parser.hpp]: No c-style cast please. Use the appropriated cast.

File: include/inja/parser.hpp [hunk 2]
```diff
@@ -315,7 +315,7 @@ class Parser {
         if (!parse_expression(tmpl)) return false;
 
         // update "previous jump"
-        if_data.prev_cond_jump = tmpl.bytecodes.size();
+        if_data.prev_cond_jump = (uint32_t)tmpl.bytecodes.size();
```
Comment (gracicot) [inline include/inja/parser.hpp]: Same

File: include/inja/parser.hpp [hunk 3]
```diff
@@ -346,7 +346,7 @@ class Parser {
 
       if (!parse_expression(tmpl)) return false;
 
-      m_loop_stack.push_back(tmpl.bytecodes.size());
+      m_loop_stack.push_back((uint32_t)tmpl.bytecodes.size());
```
Comment (gracicot) [inline include/inja/parser.hpp]: Same

File: include/inja/renderer.hpp [hunk 4]
```diff
@@ -110,7 +110,7 @@ class Renderer {
     LoopLevel& level = m_loop_stack.back();
 
     if (m_loop_stack.size() > 1) {
-      for (int i = m_loop_stack.size() - 2; i >= 0; i--) {
+      for (int i = (int)m_loop_stack.size() - 2; i >= 0; i--) {
```
Comment (gracicot) [inline include/inja/renderer.hpp]: Same

**Comment by gracicot** [comment]
What do you think about the macros @pantor ? If macros don't bother you would you prefer them at least namspaced (eg. `PANTOR_INJA_BC_SET_ARGS`) or you have something else in mind?

**Pull request closed without merging**